disttask/importinto/scheduler.go

```diff
@@ -251,8 +259,85 @@ func (s *importStepExecutor) Rollback(context.Context) error {
 	return nil
 }
 
-type writeAndIngestStepExecutor struct {
+type mergeSortStepExecutor struct {
 	scheduler.EmptySubtaskExecutor
+	taskID     int64
+	taskMeta   *TaskMeta
+	logger     *zap.Logger
+	controller *importer.LoadDataController
+	// subtask of a task is run in serial now, so we don't need lock here.
+	// change to SyncMap when we support parallel subtask in the future.
+	subtaskSortedKVMeta *external.SortedKVMeta
+}
+
+var _ execute.SubtaskExecutor = &mergeSortStepExecutor{}
+
+func (m *mergeSortStepExecutor) Init(ctx context.Context) error {
+	controller, err := buildController(&m.taskMeta.Plan, m.taskMeta.Stmt)
+	if err != nil {
+		return err
+	}
+	if err = controller.InitDataStore(ctx); err != nil {
+		return err
+	}
+	m.controller = controller
+	return nil
+}
+
+func (m *mergeSortStepExecutor) RunSubtask(ctx context.Context, subtask *proto.Subtask) (err error) {
+	logger := m.logger.With(zap.Int64("subtask-id", subtask.ID))
+	task := log.BeginTask(logger, "run subtask")
+	defer func() {
+		task.End(zapcore.ErrorLevel, err)
+	}()
+
+	sm := &MergeSortStepMeta{}
+	err = json.Unmarshal(subtask.Meta, sm)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	var mu sync.Mutex
+	m.subtaskSortedKVMeta = &external.SortedKVMeta{}
+	onClose := func(summary *external.WriterSummary) {
+		mu.Lock()
+		defer mu.Unlock()
+		m.subtaskSortedKVMeta.MergeSummary(summary)
+	}
+
+	writerID := uuid.New().String()
+	prefix := subtaskPrefix(m.taskID, subtask.ID)
+
+	return external.MergeOverlappingFiles(
+		ctx,
+		sm.DataFiles,
+		m.controller.GlobalSortStore,
+		64*1024,
+		prefix,
+		writerID,
+		256*size.MB,
```

Comment:
```suggestion
		256*size.MB,
```
change to memoryQuota?

Reply:
`merge-sort` shouldn't have the problem as in `encode&sort`, we divide subtask by kv group, i.e. data or some index, so each concurrency can use it's own writer

but i didn't loop inside